tests/utils/test_proto_json_utils.py

```diff
@@ -223,3 +235,94 @@ def test_parse_tf_serving_raises_expected_errors():
+            "string": ["a", "b", "c"],
+            "float": np.array([1.2, 2.3, 3.4], dtype=np.float32),
+            "double": np.array([1.2, 2.3, 3.4], dtype=np.float64),
+            "integer": np.array([3, 4, 5], dtype=np.int32),
+            "long": np.array([3, 4, 5], dtype=np.int64),
+            "binary": [bytes([1, 2, 3]), bytes([4, 5]), bytes([6])],
+        },
+        columns=["boolean", "string", "float", "double", "integer", "long", "binary"],
+    )
+
+    jsonable_df = pd.DataFrame(source, copy=True)
+    jsonable_df["binary"] = jsonable_df["binary"].map(base64.b64encode)
+    schema = Schema(
+        [
+            ColSpec("boolean", "boolean"),
+            ColSpec("string", "string"),
+            ColSpec("float", "float"),
+            ColSpec("double", "double"),
+            ColSpec("integer", "integer"),
+            ColSpec("long", "long"),
+            ColSpec("binary", "binary"),
+        ]
+    )
+    expected_dtypes = {
+        "boolean": DataType.boolean.to_pandas(),
+        "string": np.dtype("object"),
+        "float": DataType.float.to_pandas(),
+        "double": DataType.double.to_pandas(),
+        "integer": DataType.integer.to_pandas(),
+        "long": DataType.long.to_pandas(),
+        "binary": np.dtype("object"),
+    }
+    parsed = _dataframe_from_json(
+        jsonable_df.to_json(orient="split"), pandas_orient="split", schema=schema
+    )
+    assert parsed.equals(source)
+    parsed = _dataframe_from_json(
+        jsonable_df.to_json(orient="records"), pandas_orient="records", schema=schema
+    )
+    assert parsed.equals(source)
+    # try parsing with tensor schema
+    tensor_schema = Schema(
+        [
+            TensorSpec(np.dtype("bool"), [-1], "boolean"),
+            TensorSpec(np.dtype("str"), [-1], "string"),
+            TensorSpec(np.dtype("float32"), [-1], "float"),
+            TensorSpec(np.dtype("float64"), [-1], "double"),
+            TensorSpec(np.dtype("int32"), [-1], "integer"),
+            TensorSpec(np.dtype("int64"), [-1], "long"),
+            TensorSpec(np.dtype(bytes), [-1], "binary"),
+        ]
+    )
+    parsed = _dataframe_from_json(
+        jsonable_df.to_json(orient="split"), pandas_orient="split", schema=tensor_schema
+    )
+
+    # NB: tensor schema does not automatically decode base64 encoded bytes.
+    assert parsed.equals(jsonable_df)
+    parsed = _dataframe_from_json(
+        jsonable_df.to_json(orient="records"), pandas_orient="records", schema=tensor_schema
+    )
+
+    # NB: tensor schema does not automatically decode base64 encoded bytes.
+    assert parsed.equals(jsonable_df)
+
+    # Test parse with TesnorSchema with a single tensor
+    tensor_schema = Schema([TensorSpec(np.dtype("float32"), [-1, 3])])
+    source = pd.DataFrame(
+        {
+            "a": np.array([1, 2, 3], dtype=np.float32),
+            "b": np.array([4, 5, 6], dtype=np.float32),
+            "c": np.array([7, 8, 9], dtype=np.float32),
```

Comment:
This test will make more sense if the dtypes of the remaining inputs are not same, to ensure we can check that they were converted based on the singular tensorSpec in the Schema

Reply:
Hmm, I don't understand what do you mean? The types have to be the same since we parse them as a single type. We do make sure they're typecasted to float32, without casting they would've been read as int64. I guess I can make some of the columns floats.

Reply:
Ohh that makes sense, my bad.